compiler/hash-typecheck/src/ops/typing.rs

```diff
@@ -344,6 +367,52 @@ impl<'gs, 'ls, 'cd, 's> Typer<'gs, 'ls, 'cd, 's> {
                 // Forward to the pattern
                 self.term_of_pattern(pattern_if.pattern)
             }
+        }?;
+
+        // Copy location:
+        self.location_store_mut().copy_location(pattern_id, ty_of_pattern);
+
+        Ok(ty_of_pattern)
+    }
+
+    /// Get the parameters of the given tuple term, if possible.
+    ///
+    /// This function returns Some(..) if the term is validated and simplified,
+    /// and is a tuple term (either literal or Rt.). Otherwise it will
+    /// return None.
+    ///
+    /// This function will populate default values if it can (if the tuple is a
+    /// literal).
```

Comment:
work towards #295?

Reply:
No, unrelated. It just utilises the default value field to save arg values when converting to params. (if they are needed later on)